File: 2216-DeleteTheMiddleNodeOfALinkedList/2216-DeleteTheMiddleNodeOfALinkedList.cpp
// Last updated: 5/10/2026, 11:46:09 PM
class Solution {
    int sizE(ListNode* head){
        int count = 1;
        while(head->next){
            head = head->next;
            count++;
        }
        return count/2;
    }
public:
    ListNode* deleteMiddle(ListNode* head) {
        if(head == NULL)return NULL;
        int size = sizE(head)-1;
        if(size < 0)return NULL;
        ListNode* temp = head;
        while(size--){
            temp = temp->next;
        }
        ListNode* del = temp->next;
        temp->next = temp->next->next;
        delete del;
        return head;
    }
};

File: 2216-DeleteTheMiddleNodeOfALinkedList/2216-DeleteTheMiddleNodeOfALinkedList.java
// Last updated: 6/18/2026, 4:20:52 PM
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        if(head == null || head.next == null) return null;

        ListNode prev = null;
        ListNode slow = head;
        ListNode fast = head;
        while(fast!=null && fast.next != null){
            prev = slow;
            slow = slow.next;
            fast = fast.next.next;
        }
        prev.next = slow.next;
        return head;
    }
}